src/vcpkg/base/system.cpp

```diff
@@ -221,6 +227,32 @@ namespace vcpkg
 #endif
     }
 
+    void print_environment_variables()
+    {
+        msg::write_unlocalized_text_to_stdout(Color::none, "The following environment variables are currently set:\n");
+#if defined(_WIN32)
+        auto free = [](LPWCH p) { FreeEnvironmentStringsW(p); };
+        std::unique_ptr<WCHAR, decltype(free)> env_block(GetEnvironmentStringsW(), free);
```

Comment:
I have a mild preference for a bespoke struct with a dtor over unique_ptr since that introduces unnecessary null checks etc.

Reply:
I don't think that this one null check is really important ...

Reply:
IIRC it's more than one but it isn't enough to lose sleep over, which is why I pointed out the problem rather than saying please change it.

However:

```
struct EnvironmentStringsW {
    LPWCH strings;
    EnvironmentStringsW() : strings(GetEnvironmentStringsW()) {}
    ~EnvironmentStringsW() { FreeEnvironmentStringsW(strings); }
    EnvironmentStringsW(const EnvironmentStringsW&) = delete;
};
```

or similar isn't really much more code than you wrote here (and avoids shadowing the name free, etc.)

I'm not blocking over this suggestion though.